addons/mail/static/tests/owl/component/activity_tests.js

```diff
@@ -0,0 +1,832 @@
+        "“test summary”",
+        "should have the activity summary in activity summary div"
+    );
+});
+
+QUnit.test('activity without summary layout', async function (assert) {
+    assert.expect(5);
+
+    await this.start();
+    const activityLocalId = this.env.store.dispatch('_createActivity', {
+        activity_type_id: [1, 'Fake type'],
+    });
+    await this.createActivityComponent(activityLocalId);
+    assert.strictEqual(
+        document.querySelectorAll('.o_Activity').length,
+        1,
+        "should have activity component in DOM"
+    );
+    const activity = document.querySelector('.o_Activity');
+    assert.strictEqual(
+        activity.querySelectorAll('.o_Activity_type').length,
+        1,
+        "should have activity sidebar type"
+    );
+    assert.strictEqual(
+        activity.querySelector('.o_Activity_type').textContent.trim(),
+        "Fake type",
+        "should have the activity type display name in activity type div"
+    );
+    assert.strictEqual(
+        activity.querySelectorAll('.o_Activity_summary.o_Activity_type').length,
+        1,
+        "should have activity type as summary"
+    );
+    assert.strictEqual(
+        activity.querySelectorAll('.o_Activity_summary:not(.o_Activity_type)').length,
+        0,
+        "should not have summary"
+    );
+});
+
+QUnit.test('activity details toggle', async function (assert) {
+    assert.expect(5);
+
+    await this.start();
+    const today = new Date();
+    const tomorrow = new Date();
+    tomorrow.setDate(today.getDate() + 1);
+    const activityLocalId = this.env.store.dispatch('_createActivity', {
+        activity_type_id: [1, 'Fake type'],
+        create_date: date_to_str(today),
+        create_uid: [1, 'Admin'],
+        date_deadline: date_to_str(tomorrow),
+        state: 'planned',
+    });
+    await this.createActivityComponent(activityLocalId);
+    assert.strictEqual(
+        document.querySelectorAll('.o_Activity').length,
+        1,
+        "should have activity component in DOM"
+    );
+    const activity = document.querySelector('.o_Activity');
+    assert.strictEqual(
+        activity.querySelectorAll('.o_Activity_details').length,
+        0,
+        "activity should have no details"
+    );
+    assert.strictEqual(
+        activity.querySelectorAll('.o_Activity_detailsButton').length,
+        1,
+        "activity should have a details button"
+    );
+
+    await activity.querySelector('.o_Activity_detailsButton').click();
+    await afterNextRender();
+    // await pause();
+    assert.strictEqual(
+        activity.querySelectorAll('.o_Activity_details').length,
+        1,
+        "activity should have details"
+    );
+
+    await activity.querySelector('.o_Activity_detailsButton').click();
+    await afterNextRender();
+    assert.strictEqual(
+        activity.querySelectorAll('.o_Activity_details').length,
+        0,
+        "activity should have no details"
```

Comment:
it's much nicer for assertion comment to give some context, like here it tells details are no longer visible after clicking on details button.